### ISSUE TITLE
Change API for feature website

### ISSUE THREAD

**Issue by futag** [JustHumanz/Go-Simp#265]
I have some changes api following change new website
From
``` json
 {
  "BiliBili":{...},
  "EnName":"Sakura Miko",
  "Fanbase":null,
  "GroupID":5,
  "GroupName":"Hololive",
  "ID":56,
  "JpName":"さくらみこ",
  "NickName":"Miko-chi",
  "Region":"JP",
  "Twitch":{...},
  "Twitter":{...},
  "Youtube":{...}
}
```
 To
``` Json
 {
  "BiliBili":{...},
  "EnName":"Sakura Miko",
  "Fanbase":null,
  "Group":{
    "GroupIcon":"https://cdn.humanz.moe/holo.png",
    "GroupName":"Hololive",
    "ID":5,
  },
  "ID":56,
  "JpName":"さくらみこ",
  "NickName":"Miko-chi",
  "Region":"JP",
  "Twitch":{...},
  "Twitter":{...},
  "Youtube":{...}
}
```

TO DO LIST

- [x] Add Group fill for replaceing "GroupID", and  "GroupName" like above
- [x] Add array in LiveURL when live multi steam (like youtube and bilibili) instead string (or make 2 referent fill)
- [x] Support **live=true** in all vtubers

**Comment by JustHumanz** [comment]
okey

**Comment by JustHumanz** [comment]
the Reset API V2 already deployed,the path still same just add `/v2` after the domain name, here some example

https://api.humanz.moe/v2/members/  
https://api.humanz.moe/v2/groups/

just let me know if you need something

**Comment by futag** [comment]
`IsLive` change array to object to easy fliter (with name state)

**Comment by futag** [comment]
and `v2` have duplicate data

**Comment by JustHumanz** [comment]
on check

**Comment by JustHumanz** [comment]
already updated,you can check it

**Comment by futag** [comment]
![image](https://user-images.githubusercontent.com/21020026/156082637-a437d0aa-3fa3-43f8-b755-b78b7a6644dc.png)
Some data unstable

**Comment by JustHumanz** [comment]
unstable?

**Comment by futag** [comment]
![image](https://user-images.githubusercontent.com/21020026/156103506-f9e0b551-cf9f-4bf6-8b98-75cffd5a3b9e.png)
Old api
![image](https://user-images.githubusercontent.com/21020026/156103534-361baf32-2d9b-4207-ab4d-fb2fbd25ae8c.png)
New Api

so some data sometimes doesn't appear, but sometimes all data appears

**Comment by JustHumanz** [comment]
already fixed in e5167a08f8aee70d953a0ed7eac07686181e7727

**Comment by futag** [comment]
Where `Bilibili.Followers`?
and some vtuber members in groups sometimes the data doesn't match the original API, but groups and all members match

Holo members in new API ![image](https://user-images.githubusercontent.com/21020026/156297671-dba9710a-ec37-43e8-944e-e710407fa0e5.png)
Holo members in old API ![image](https://user-images.githubusercontent.com/21020026/156297828-7ee223b8-5c5c-47ce-a673-3a9ff7d23ef3.png)

**Comment by JustHumanz** [comment]
lmao,i'm forget it

doesn't match? is that you use `groupid` parameter?

**Comment by JustHumanz** [comment]
`Bilibili.Followers` already add

**Comment by futag** [comment]
yes

**Comment by futag** [comment]
i use groupid, but data not match the original API

**Comment by JustHumanz** [comment]
okey,let me check it

**Comment by futag** [comment]
but small change for get unmatch

**Comment by JustHumanz** [comment]
just found the root case 

![Screenshot from 2022-03-02 21-57-07](https://user-images.githubusercontent.com/43176061/156390997-b708a1c6-edbe-4d2f-9b0b-fbb2ef17b3b3.png)

i use goroutine when filtering members data then this error show up, already fix it

**Comment by futag** [comment]
To do list: final one before (merging)

- [x] hide useless data, and only show in singular/single data (all TotalVideos, fanart and lewd hastag, YoutubeChannels in group)
- [x] In Single data, add cover art in youtube, twitch, and bilibili

**Comment by JustHumanz** [comment]
1. did you mean remove the TotalVideos, fanart and lewd hastag, YoutubeChannels in group from json?
so it's will become like this?
```
[
    {
       "ID":14,
       "NickName":"Parerun",
       "EnName":"Hanamaru Hareru",
       "JpName":"花丸はれる",
       "Region":"JP",
       "Fanbase":"",
       "Status":"Active",
       "BiliBili":{
          "Avatar":"https://i0.hdslb.com/bfs/face/7328c4d2bd7f59a10be61589d3595880a8ebcff5.jpg",
          "Followers":432208,
          "LiveID":21547895,
          "SpaceID":441381282,
          "ViwersCount":27096695
       },
       "Youtube":{
          "Avatar":"https://yt3.ggpht.com/ytc/AKedOLQOe0o5KY82FMZ27lVPWFtCWXZWOAVwPUqlHs-R=s800-c-k-c0x00ffffff-no-rj",
          "Subscriber":64200,
          "ViwersCount":4687140,
          "YoutubeID":"UCyIcOCH-VWaRKH9IkR8hz7Q"
       },
       "Twitter":{
          "Followers":24674,
          "Username":"hanamaruhareru"
       },
       "Twitch":null,
       "Group":{
          "ID":10,
          "IconURL":"https://cdn.humanz.moe/404.jpg",
          "GroupName":"independent",
       },
       "IsLive":{
          "BiliBili":null,
          "Twitch":null,
          "Youtube":null
       }
    }
 ]
```

The cover art?
do you mean the cover of channel?

![Screenshot 2022-03-03 at 13-47-46 鹿乃 Kano official channel](https://user-images.githubusercontent.com/43176061/156511716-3b7e815f-e7e6-4c8a-a424-b0b596164e11.png)

**Comment by futag** [comment]
yes when exist every platform

**Comment by futag** [comment]
> 1. did you mean remove the TotalVideos, fanart and lewd hastag, YoutubeChannels in group from json?
>        so it's will become like this?
> 
> 
> ```
> [
>     {
>        "ID":14,
>        "NickName":"Parerun",
>        "EnName":"Hanamaru Hareru",
>        "JpName":"花丸はれる",
>        "Region":"JP",
>        "Fanbase":"",
>        "Status":"Active",
>        "BiliBili":{
>           "Avatar":"https://i0.hdslb.com/bfs/face/7328c4d2bd7f59a10be61589d3595880a8ebcff5.jpg",
>           "Followers":432208,
>           "LiveID":21547895,
>           "SpaceID":441381282,
>           "ViwersCount":27096695
>        },
>        "Youtube":{
>           "Avatar":"https://yt3.ggpht.com/ytc/AKedOLQOe0o5KY82FMZ27lVPWFtCWXZWOAVwPUqlHs-R=s800-c-k-c0x00ffffff-no-rj",
>           "Subscriber":64200,
>           "ViwersCount":4687140,
>           "YoutubeID":"UCyIcOCH-VWaRKH9IkR8hz7Q"
>        },
>        "Twitter":{
>           "Followers":24674,
>           "Username":"hanamaruhareru"
>        },
>        "Twitch":null,
>        "Group":{
>           "ID":10,
>           "IconURL":"https://cdn.humanz.moe/404.jpg",
>           "GroupName":"independent",
>        },
>        "IsLive":{
>           "BiliBili":null,
>           "Twitch":null,
>           "Youtube":null
>        }
>     }
>  ]
> ```

yes
when show a member,  showing all data, like https://api.humanz.moe/v2/members/76

**Comment by JustHumanz** [comment]
ah ok,got it man

but for cover art it's will take some time since the bot not collect the cover art by default,so i need rework some module

**Comment by JustHumanz** [comment]
bad news

twitch API don't support get channel banner and look like they will not add it on their API

https://discuss.dev.twitch.tv/t/twitch-api-user-profile-banner/24463/2?u=justhumanz

**Comment by JustHumanz** [comment]
so yeah,we will go without twitch banner

**Comment by futag** [comment]
done?

**Comment by JustHumanz** [comment]
Still on progress

**Comment by futag** [comment]
some avatar in bilibili null and fanart bilibili null in v2 (v1 normal)

**Comment by futag** [comment]
and lewd twitter showing banner link

**Comment by JustHumanz** [comment]
Already add `banner` & fixed
pls check it

**Comment by futag** [comment]
Full data on single vtuber not show in new data api 

https://api.humanz.moe/v2/members/188

probably, some vtuber cannot have cover like haachama

**Comment by futag** [comment]
Data minutes ago
![Screenshot 2022-03-07 at 09-19-39 https api humanz moe](https://user-images.githubusercontent.com/21020026/156956798-db15a232-3100-4166-a0ee-dcc32f3e874e.png)
Data few seconds
![Screenshot 2022-03-07 at 09-19-54 https api humanz moe](https://user-images.githubusercontent.com/21020026/156956856-a1c1b5e4-90cb-4582-a90b-f28471b8b117.png)
but, sometimes all data appears

**Comment by futag** [comment]
Maybe it's because there are many versions of the data load, so that single data shows what data is displayed which has multiple versions

**Comment by JustHumanz** [comment]
wait, let me check it

**Comment by futag** [comment]
try open single data, then open all vtuber data, then open single data again (refresh)

**Comment by JustHumanz** [comment]
already solved, that happen because i use `delete` func and that makes the data will be deleted from the struct 

![Screenshot from 2022-03-07 13-18-48](https://user-images.githubusercontent.com/43176061/156978622-1b2386e6-2ee1-42ab-8d79-b6937df32e1d.png)

**Comment by futag** [comment]
![image](https://user-images.githubusercontent.com/21020026/156995976-bedf9a02-11e4-4945-8572-6a2722421ab3.png)
Somting worng with vtuber without banner (like haachama)

**Comment by JustHumanz** [comment]
Because hacama don't have any banner in her channel

![Screenshot 2022-03-07 at 15-51-30 HAACHAMA Ch 赤井はあと](https://user-images.githubusercontent.com/43176061/156998575-be76cba1-ad92-4628-8004-e809d130a704.png)

**Comment by futag** [comment]
but `=s1200` still exist, even if there is no banner

**Comment by JustHumanz** [comment]
the data already refreshed, `=s1200` already gone

**Comment by futag** [comment]
https://api.humanz.moe/v2/members/1
Data not return 404
![image](https://user-images.githubusercontent.com/21020026/157168149-bd9b010a-8bcb-4909-9b22-e38d077bf6bb.png)

**Comment by futag** [comment]
All member hololive members is japan region
https://api.humanz.moe/v2/members/?groupid=5
https://api.humanz.moe/members/?groupid=5

**Comment by futag** [comment]
and some members in worng region

**Comment by JustHumanz** [comment]
all already fixed

**Comment by futag** [comment]
Bilibili liveID worng id (same like space id)

**Comment by JustHumanz** [comment]
fixed a01aa7bb76a7023822413d6b8d147310e5d10294
